test/executor.test.js

```diff
@@ -906,6 +906,62 @@ describe('executor', function() {
     });
   });
 
+  describe('when booting with lazy connect', function() {
+    var sampleInstruction = someInstructions({
+      dataSources: {
+        lazyConnector: {
+          connector: 'testLazyConnect',
+          name: 'lazyConnector',
+        },
+      },
+    });
+    var connectTriggered = true;
```

Comment:
Please do reset it at the beginning. If it happens that the code in your custom connector does not get called by some weird error that is swallowed, or perhaps because the connector API changed and our mocked connector is no longer valid, then test will see the result of the previous call, which is not right.

IMO, the most correct solution is to:
 - reset the variable in `beforeEach` hook to `undefined` or `null`
 - inside the connector, set the variable to true/false depending on the value of `lazyConnect`, as you shown in the code snippet above